internal/cmd/plugin/color.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/logrusorgru/aurora/v4"
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+)
+
+// ConfigureColor renews aurora.DefaultColorizer based on flags and TTY
+func ConfigureColor(cmd *cobra.Command) error {
+	return configureColor(cmd, term.IsTerminal(int(os.Stdout.Fd())))
+}
+
+func configureColor(cmd *cobra.Command, isTTY bool) error {
+	colorFlag, err := cmd.Flags().GetString("color")
+	if err != nil {
+		return err
+	}
+
+	var shouldColorize bool
+	switch colorFlag {
```

Comment:
In my opinion, `colorFlag` is not a string but a custom type representing an enumeration. 
A similar concept is implemented in the [snapshot owner reference](https://github.com/cloudnative-pg/cloudnative-pg/blob/0d27d7603739729d2e6cf3d178ea5d604ca5de6f/api/v1/cluster_types.go#L130) field.

Once we have a custom type with the relative constants, the following `switch` statement could use them instead of these strings.

The custom type could also implement `pflag.Value` encapsulating validation. 

[This comment](https://github.com/spf13/pflag/issues/236#issuecomment-931600452) contains a really nice example.

Reply:
Hoo an Enum with the proper auto-complete :D just saying!

Reply:
Thanks, Leo! I believe what you mean is to implement a custom type specifically for the Color flag, not a generic enumeration type.
Please check here https://github.com/cloudnative-pg/cloudnative-pg/pull/4775/files#diff-f6d12283f95d10d3d682413241cab651cec5e25f0c63d84ee42b292a717a5a53R28-R54

Reply:
@YanniHu1996 That's exactly what I meant with: "custom type representing an enumeration"